api/src/main/java/io/opencensus/trace/SpanBuilder.java

```diff
@@ -223,6 +224,60 @@ public final Scope startScopedSpan() {
     return new ScopedSpanHandle(startSpan());
   }
 
+  /**
+   * Runs the given {@code runnable} with the newly created {@code Span} as current in the Context,
+   * and ends the {@code Span} after the {@code runnable} is run.
+   *
+   * <p>This is equivalent with following code:
+   *
+   * <pre>{@code
+   * Span span = tracer.spanBuilder("MyRunnableSpan").startSpan();
+   * try {
+   *   tracer.wrap(span, myRunnable).run();
+   * } finally {
+   *   span.end();
+   * }
+   * }</pre>
+   *
+   * @param runnable the {@code Runnable} to run in the {@code Span}.
+   */
+  public final void startSpanAndRun(Runnable runnable) {
+    Span span = startSpan();
```

Comment:
One choice here is whether to keep adding methods or do something like Spans.run(spanBuilder, Runnable)

Reply:
I think both are valid and nice solutions. I think if we have a builder it makes sense to have them here. In languages where the Builder pattern is not common your suggestion makes a lot of sense.